### PR TITLE
Verilog: `convert_range` now returns range

### PR DESCRIPTION
**Pull request by kroening**
The `convert_range` method now returns the range, as opposed to assigning to two reference parameters.

### CODE REVIEW

File: src/verilog/verilog_elaborate_type.cpp [hunk 1]
```diff
@@ -44,12 +44,10 @@ array_typet verilog_typecheck_exprt::convert_unpacked_array_type(
...
-    mp_integer msb, lsb;
-    convert_range(range_expr, msb, lsb);
-    big_endian = (lsb > msb);
-    size = (big_endian ? lsb - msb : msb - lsb) + 1;
-    DATA_INVARIANT(size >= 0, "array size must not be negative");
-    offset = big_endian ? msb : lsb;
+    auto range = convert_range(range_expr);
+    big_endian = (range.lsb > range.msb);
+    size = range.length();
+    offset = big_endian ? range.msb : range.lsb;
```
Comment (tautschnig) [inline src/verilog/verilog_elaborate_type.cpp:50]:
```suggestion
    offset = range.smallest_index();
```

File: src/verilog/verilog_interfaces.cpp [hunk 2]
```diff
@@ -163,14 +163,14 @@ void verilog_typecheckt::interface_inst(
...
+  const exprt &range_expr = static_cast<const exprt &>(op.find(ID_range));
 
-  mp_integer msb, lsb;
-  
-  if(range.is_nil() || range.id().empty())
-    msb=lsb=0;
+  ranget range;
+
+  if(range_expr.is_nil() || range_expr.id().empty())
+    range.msb = range.lsb = 0;
```
Comment (tautschnig) [inline src/verilog/verilog_interfaces.cpp:171]: Shouldn't there be a default constructor of `ranget` that does this?

File: src/verilog/verilog_typecheck.cpp [hunk 3]
```diff
@@ -203,18 +203,19 @@ Function: verilog_typecheckt::typecheck_builtin_port_connections
...
-  exprt &range=static_cast<exprt &>(inst.add(ID_range));
+  exprt &range_expr = static_cast<exprt &>(inst.add(ID_range));
 
-  mp_integer msb, lsb;
-  
-  if(range.is_nil() || range.id()=="")
-    msb=lsb=0;
+  ranget range;
+
+  if(range_expr.is_nil() || range_expr.id() == "")
```
Comment (tautschnig) [inline src/verilog/verilog_typecheck.cpp:210]: Should use `.empty()` or `== irep_idt()`, see also comment above.

File: src/verilog/verilog_typecheck.cpp [hunk 4]
```diff
@@ -203,18 +203,19 @@ Function: verilog_typecheckt::typecheck_builtin_port_connections
...
+    range.msb = range.lsb = 0;
   else
-    convert_range(range, msb, lsb);
+    range = convert_range(range_expr);
+
+  if(range.lsb > range.msb)
+    std::swap(range.lsb, range.msb);
```
Comment (tautschnig) [inline src/verilog/verilog_typecheck.cpp:216]: Why is this here (and just here) the right thing to do?

File: src/verilog/verilog_typecheck_expr.cpp [hunk 5]
```diff
@@ -2101,19 +2101,20 @@ Function: verilog_typecheck_exprt::convert_range
...
   {
     throw errort().with_location(range.source_location())
       << "range expected to have two operands";
   }
 
-  msb = convert_integer_constant_expression(to_binary_expr(range).op0());
-  lsb = convert_integer_constant_expression(to_binary_expr(range).op1());
+  ranget result;
+  result.msb = convert_integer_constant_expression(to_binary_expr(range).op0());
+  result.lsb = convert_integer_constant_expression(to_binary_expr(range).op1());
```
Comment (tautschnig) [inline src/verilog/verilog_typecheck_expr.cpp:2115]: There should be a 2-argument constructor for `ranget`.

**Review by tautschnig — APPROVED** [review]

File: src/verilog/verilog_interfaces.cpp (hunk 2 above)
Comment (kroening) [reply src/verilog/verilog_interfaces.cpp:171]: Considered it, but would you remember whether msb is the first argument, or lsb?

Comment (kroening) [reply src/verilog/verilog_interfaces.cpp:171]: I'll go for msb, lsb.

File: src/verilog/verilog_typecheck.cpp (hunk 4 above)
Comment (kroening) [reply src/verilog/verilog_typecheck.cpp:216]: It's probably broken. I'll investigate.
